selenium_tests/questionnaires_all/test_questionnaire_functionality.py

```diff
@@ -165,11 +165,116 @@ def test_survey_back(self):
         results_page.go_back()
         current_page = QuestionnairePage(self.selenium)
         self.assertIn(survey01.title, current_page.title)
-        
-        
 
+    def test_skip_question(self):
+        survey01 = SurveyFactory(
+            parent=self.section01, 
+            owner=self.user,
+            allow_multiple_submissions = True,
+            multi_step = True,
+            thank_you_text="[{\"type\": \"paragraph\", \"value\": \"<p>Thankyou for completing the survey</p>\"}]"
+            )
 
+        SurveyFormFieldFactory(
+            page=survey01,
+            pk= 1,
+            sort_order= 0,
+            required=True,
+            choices= "A|B|C",
+            clean_name= "question_1",
+            label= "XXXX",
+            field_type= "radio",
+            admin_label= "q1",
+            skip_logic= "[{\"type\": \"skip_logic\", \"value\": {\"choice\": \"A\", \"skip_logic\": \"next\", \"question\": null}, \"id\": \"60a035c2-7e67-4c41-90c7-8be3a94faefb\"}, {\"type\": \"skip_logic\", \"value\": {\"choice\": \"B\", \"skip_logic\": \"question\", \"question\": 3}, \"id\": \"6b470d57-a397-4eec-a747-d62652fc56c2\"}, {\"type\": \"skip_logic\", \"value\": {\"choice\": \"C\", \"skip_logic\": \"end\", \"question\": null}, \"id\": \"8d0ccb1e-6d44-4d95-aeab-ea66512f5c68\"}]",
```

Comment:
There must be a better way to deal with this field - as it stands it's not very readable.

One method would be to define the skip logic using Python lists and dicts and then converting them to JSON using the json module:

```
import json
skip_logic = [ {'type': 'skip_logic', ...} ]
json = json.dumps(skip_logic)
```

Reply:
have implemented as suggested